### PR TITLE
Implement Debug for (ed25519/secp256k1)::(Keypair|SecretKey)

### PR DESCRIPTION
**Pull request by c410-f3r**
Sidenote: I see a bunch of deprecated functions, Rust 2015 stuff and `std::mem::uninitialized`. Is anyone going to update this project?

### CODE REVIEW

**Comment by tomaka** [comment]
We preferably don't want secret keys to be accidentally printed on stdout, ever.
If we are to implement `Debug`, let's have a dummy manual implementation instead.

**Comment by c410-f3r** [comment]
> If we are to implement Debug, let's have a dummy manual implementation instead.

Thank you. Can it be a bunch of zeros?

File: core/src/identity/secp256k1.rs [hunk 1]
```diff
@@ -51,6 +52,12 @@ impl Keypair {
     }
 }
 
+impl fmt::Debug for Keypair {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "Keypair {{ secret: \"Secret Key\", public: \"{:?}\" }}", &self.public)
```
Comment (tomaka) [inline core/src/identity/secp256k1.rs]: The `Formatter` has helpers for outputting structs:

```suggestion
        f.debug_struct("Keypair").field("public", &self.public).finish()
```

File: core/src/identity/secp256k1.rs [hunk 2]
```diff
@@ -70,6 +77,12 @@ impl From<Keypair> for SecretKey {
 #[derive(Clone)]
 pub struct SecretKey(secp256k1::SecretKey);
 
+impl fmt::Debug for SecretKey {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "Secret Key")
```
Comment (tomaka) [inline core/src/identity/secp256k1.rs]:
```suggestion
        write!(f, "SecretKey")
```

File: core/src/identity/ed25519.rs [hunk 3]
```diff
@@ -66,6 +67,16 @@ impl Keypair {
     }
 }
 
+impl fmt::Debug for Keypair {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(
```
Comment (tomaka) [inline core/src/identity/ed25519.rs]:
```suggestion
        f.debug_struct("Keypair").field("public", &self.0.public).finish()
```

**Comment by tomaka** [comment]
> I never had the change to tell this but I would like thank you for all the awesome work you did for Rust, specially in the gaming ecosystem

Thank you!

**Comment by c410-f3r** [comment]
Updated

**Pull request merged**